local/compose/create.go

```diff
@@ -129,6 +135,30 @@ func prepareNetworks(project *types.Project) {
 	}
 }
 
+func prepareNetworkFrom(p *types.Project) error {
+outLoop:
+	for i := range p.Services {
+		dependency := getNetworkFrom(p.Services[i].NetworkMode)
+		if dependency != "" {
+			if p.Services[i].DependsOn == nil {
+				p.Services[i].DependsOn = make(types.DependsOnConfig)
+			}
+
+			for _, service := range p.Services {
+				if service.Name == dependency {
+					p.Services[i].DependsOn[service.Name] = types.ServiceDependency{
+						Condition: types.ServiceConditionStarted,
+					}
+					continue outLoop
+				}
+			}
+
+			return fmt.Errorf("service %q not found for network_mode 'service:%s'", dependency, dependency)
```

Comment:
such a check should take place in compose-go

Reply:
Included in https://github.com/compose-spec/compose-go/pull/109